src/Compilers/Server/VBCSCompilerTests/HackUtil.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Pipes;
+using System.Reflection;
+using System.Net.Sockets;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CompilerServer.UnitTests
+{
+    /// <summary>
+    /// This is a HACK that allows you to get at the underlying Socket for a given NamedPipeServerStream 
```

Comment:
>This is a HACK that [](start = 8, length = 19)

Consider removing, and renaming the class, perhaps something like "SocketTestUtils".

Reply:
Will rename. Even though it's a giant hack it's important to keep because it makes it infinitely easier to diagnose the leaks that were leading to the finalizer deadlock.